controllers/authController.js

```diff
@@ -0,0 +1,75 @@
+const Users = require('../models/Users');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+
+// Sign Up
+const signUp = async (req , res)  =>{
+    const {name , email , username , password , password2 } = req.body;
+    //Validation
+    if(!name , !email , !username , !password, !password2 ){
+        res.status(422).json({error:"Please fill in all field"})
+    }
+
+    if(password !== password2){
+        res.status(422).json({error:"password does not match"})
+
+    }
+
+    if(password < 6){
+       res.status(422).json({error:"password must not be less than six characters"})
+    }
+    
+    await  Users.findOne({email : email})
+    .then(user =>{
+      if(user){
+        res.status(422).json({error:"Email already registered"})
```

Comment:
You are not using the `async-await` pattern correctly

Reply:
Ok